src/backend/distributed/utils/maintenanced.c

```diff
@@ -309,31 +296,45 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		 * tasks should do their own time math about whether to re-run checks.
 		 */
 
-		if (secondsSincePrevStatsCollection >= STATISTICS_COLLECTION_INTERVAL ||
-			(prevStatsCollectionFailed &&
-			 secondsSincePrevStatsCollection >= STATISTICS_COLLECTION_RETRY_INTERVAL))
-		{
 #ifdef HAVE_LIBCURL
-			if (EnableStatisticsCollection)
-			{
-				MemoryContext statsCollectionContext =
-					AllocSetContextCreate(CurrentMemoryContext,
-										  "StatsCollection",
-										  ALLOCSET_DEFAULT_MINSIZE,
-										  ALLOCSET_DEFAULT_INITSIZE,
-										  ALLOCSET_DEFAULT_MAXSIZE);
-				MemoryContext oldContext =
-					MemoryContextSwitchTo(statsCollectionContext);
+		if (EnableStatisticsCollection &&
+			difftime(time(NULL), prevStatsCollection) >= statsCollectionInterval)
+		{
+			bool statsCollectionSuccess = false;
+			StartTransactionCommand();
```

Comment:
It might make sense to call `InvalidateMetadataSystemCache()` at the start of the `if` statement just before starting the transaction, and do that again in the deadlock detection. Otherwise, we take cached OIDs from the call home into the deadlock detection, which could cause trouble.